public/js/reducers/tabs.js

```diff
@@ -47,4 +60,5 @@ function getTabId(tab) {
   return id;
 }
 
+const update = setupReducer(initialState, { addTabs, selectTab });
```

Comment:
I actually quite like this pattern for a couple of reasons:
* it enumerates the supported handlers in a more succinct fashion than the case statement
* makes it possible to invoke a single handler for testing (maybe useful)
* avoids the smell of one large function
* makes it easy to document each function separately